app/ui-utils/client/lib/MessageAction.js

```diff
@@ -172,13 +172,21 @@ Meteor.startup(async function() {
 				reply: msg._id,
 			});
 		},
-		condition({ subscription, room }) {
+		condition({ subscription, room, msg, u }) {
 			if (subscription == null) {
 				return false;
 			}
 			if (room.t === 'd' || room.t === 'l') {
 				return false;
 			}
+
+			// Check if we already have a DM started with the message user (not ourselves) or we can start one
+			const dmRoom = Rooms.findOne({ _id: [u._id, msg.u._id].sort().join('') });
+			const canAccessDM = dmRoom && Subscriptions.findOne({ rid: dmRoom._id, 'u._id': u._id });
```

Comment:
Maybe before doing multiple `findOne`s, we can check if the user lack the permission, so we can avoid calling these methods when we don't need it.

Reply:
Are you suggesting something like this?

```
// Don't show reply option if we're messaging someone else and we don't have
// permission to start a new DM with them
if (u._id !== msg.u._id && !hasPermission('create-d')) {
  const dmRoom = Rooms.findOne({ _id: [u._id, msg.u._id].sort().join('') });
  if (!dmRoom || !Subscriptions.findOne({ rid: dmRoom._id, 'u._id': u._id })) {
    return false;
  }
}
```

Reply:
@MartinSchoeler are you suggesting something like :point_up: ?

Reply:
@MartinSchoeler thoughts RE my suggested fix?

Reply:
hi @MartinSchoeler still awaiting your feedback.